easybuild/tools/filetools.py

```diff
@@ -658,17 +674,32 @@ def apply_patch(patch_file, dest, fn=None, copy=False, level=None):
         _log.debug("Changing to directory %s" % adest)
     except OSError, err:
         raise EasyBuildError("Can't change to directory %s: %s", adest, err)
-        return
 
-    patch_cmd = "patch -b -p%d -i %s" % (patch_level, apatch)
+    patch_cmd = "patch -b -p%s -i %s" % (patch_level, apatch)
     result = run.run_cmd(patch_cmd, simple=True)
     if not result:
         raise EasyBuildError("Patching with patch %s failed", patch_file)
-        return
 
     return result
 
 
+def apply_regex_substitutions(path, regex_subs):
+    """Apply specified list of regex substitutions."""
+
+    # early exit in 'dry run' mode
+    if build_option('extended_dry_run'):
+        dry_run_msg("patching file %s" % path, silent=build_option('silent'))
+        return
+
+    for i, (regex, subtxt) in enumerate(regex_subs):
+        regex_subs[i] = (re.compile(regex), subtxt)
```

Comment:
this function currently only supports patching by lines, so no multiline patching support yet (it's rather rare to require that at runtime)

we can enhance this function later to support that